tools/NetCoreCsProjSync/NetCoreCsProjSync/NetCoreCsProjGenerator.cs

```diff
@@ -0,0 +1,304 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Serialization;
+using NetCoreCsProjSync.NewModel;
+using NetCoreCsProjSync.OldModel;
+
+namespace NetCoreCsProjSync
+{
+    internal static class NetCoreCsProjGenerator
+    {
+        private const string CsProjExtension = @".csproj";
+        private const string CsProjFilter = @"*" + CsProjExtension;
+        private const string NetCoreDenoter = @".Netcore";
+        private const string NetCoreCsProjExtension = NetCoreDenoter + CsProjExtension;
+        private const string NetCoreFilter = @"*" + NetCoreCsProjExtension;
+
+        private const string CommandsFilter = @"Commands.*";
+        private const string TestFolderDenoter = @".Test";
+        private const string TestFolderDenoter2 = @".UnitTest";
+
+        // https://stackoverflow.com/a/25245678/294804
+        public static IEnumerable<string> GetProjectFolderPaths(string rmPath, bool ignoreExisting = false) =>
+            Directory.EnumerateDirectories(rmPath).SelectMany(md =>
+                Directory.EnumerateDirectories(md, CommandsFilter).Where(pd =>
+                    !pd.EndsWith(TestFolderDenoter) && !pd.EndsWith(TestFolderDenoter2) &&
+                    Directory.EnumerateFiles(pd, CsProjFilter, SearchOption.TopDirectoryOnly).Any() &&
+                    (ignoreExisting || !Directory.EnumerateFiles(pd, NetCoreFilter, SearchOption.TopDirectoryOnly).Any())));
+
+        public static IEnumerable<string> GetDesktopFilePaths(IEnumerable<string> projectPaths) =>
+            projectPaths.Select(pd => Directory.EnumerateFiles(pd, CsProjFilter, SearchOption.TopDirectoryOnly).First(f => !f.Contains(NetCoreCsProjExtension)));
+
+        public static string ConvertDesktopToNetCorePath(string desktopPath) =>
+            Path.Combine(Path.GetDirectoryName(desktopPath), Path.GetFileNameWithoutExtension(desktopPath) + NetCoreCsProjExtension);
+
+        public static IEnumerable<OldProjectDefinition> GetDesktopDefinitions(IEnumerable<string> desktopFilePaths)
+        {
+            var serializer = new XmlSerializer(typeof(OldProjectDefinition));
+            foreach (var path in desktopFilePaths)
+            {
+                using (var fileStream = new FileStream(path, FileMode.Open))
+                {
+                    var definition = (OldProjectDefinition) serializer.Deserialize(fileStream);
+                    definition.FilePath = path;
+                    yield return definition;
+                }
+            }
+        }
+
+        public static readonly List<string> ModuleSkipList = new List<string>
+        {
+            "Aks",
+            "AnalysisServices.Dataplane",
+            "Compute.ManagedService",
+            "Profile",
+            "RecoveryServices.Backup.Logger",
+            "Tags"
+        };
+
+        public static readonly Dictionary<string, List<string>> ModuleMap = new Dictionary<string, List<string>>
```

Comment:
Can we make this data external, so we don't have to edit the code to add a new module or remove a module from the skip list?

Reply:
Fixed. Allowed both `ModuleMap` and `ModuleSkipList` to be loaded from .json files.